symbolic-base/src/ZkFold/Algebra/Class.hs

```diff
@@ -248,16 +248,21 @@ class FromConstant a a => AdditiveSemigroup a where
   double :: a -> a
   double a = a + a
 
+-- | A class of types with a neutral element with respect to addition.
+-- Split into a separate class since defining addition might need additional
+-- (undesirable) constraints.
+class Zero a where
```

Comment:
Why was it necessary to separate this class from `AdditiveMonoid`?
And perhaps we need to separate `One` from `MultiplicativeMonoid` as well for consistency?